docs/sources/alerting/manage-notifications/template-notifications/write-templates.md

```diff
@@ -0,0 +1,252 @@
+
+```
+{{ range .Alerts }}
+{{ $alert := . }}
+{{ range .Labels.SortedPairs }}
+The name of the label is {{ .Name }}, and the value is {{ .Value }}
+{{/* this works because $alert refers to the value of dot inside the first range */}}
+{{ $alert.SortedPairs }}
+{{ end }}
+{{ end }}
+```
+
+## If statements
+
+You can use if statements in templates. For example, to print `There are no alerts` if there are no alerts in `.Alerts` you would write the following:
+
+```
+{{ if .Alerts }}
+{{ range .Alerts }}
+{{ .Labels }}
+{{ end }}
+{{ else }}
+There are no alerts
+{{ end }}
+```
+
+## With
+
+With is similar to if statements, however unlike if statements, `with` updates dot to refer to the value of the with:
+
+```
+{{ with .Alerts }}
+{{ range . }}
+{{ .Labels }}
+{{ end }}
+{{ else }}
+There are no alerts
+{{ end }}
+```
+
+## Indentation
+
+You can use indentation, both tabs and spaces, and line breaks, to make templates more readable:
+
+```
+{{ range .Alerts }}
+  {{ range .Labels.SortedPairs }}
+    {{ .Name }} = {{ .Value }}
+  {{ end }}
+{{ end }}
+```
+
+However, indentation in the template will also be present in the text. Next we will see how to remove it.
+
+## Remove spaces and line breaks
+
+In text/template use `{{-` and `-}}` to remove leading and trailing spaces and line breaks.
+
+For example, when using indentation and line breaks to make a template more readable:
+
+```
+{{ range .Alerts }}
+  {{ range .Labels.SortedPairs }}
+    {{ .Name }} = {{ .Value }}
+  {{ end }}
+{{ end }}
+```
+
+The indentation and line breaks will also be present in the text:
+
+```
+    alertname = "Test"
+
+    grafana_folder = "Test alerts"
+```
+
+You can remove the indentation and line breaks from the text changing `}}` to `-}}` at the start of each range:
+
+```
+{{ range .Alerts -}}
+  {{ range .Labels.SortedPairs -}}
+    {{ .Name }} = {{ .Value }}
+  {{ end }}
+{{ end }}
+```
+
+The indentation and line breaks in the template are now absent from the text:
+
+```
+alertname = "Test"
+grafana_folder = "Test alerts"
+```
+
+## Comments
+
+You can add comments with `{{/*` and `*/}}`:
+
+```
+{{/* This is a comment */}}
+```
+
+To prevent comments from adding line breaks use:
+
+```
+{{- /* This is a comment with no leading or trailing line breaks */ -}}
+```
+
+## Defining a template
+
+You can define templates using `define` and the name of the template in double quotes. You should not define templates with the same name as other templates, including default templates such as `__subject`, `__text_values_list`, `__text_alert_list`, `default.title` and `default.message`. Where a template has been created with the same name as a default template, or a template in another message template, Grafana might use either template. Grafana does not prevent, or show an error message, when there are two or more templates with the same name.
+
+```
+{{ define "print_labels" }}
+{{ end }}
+```
+
+## Executing a template
```

Comment:
```suggestion
## Execute a template
```

Reply:
Should it be plural `Execute templates`?

Reply:
yea sounds good